grails-app/services/org/pih/warehouse/data/DataService.groovy

```diff
@@ -803,9 +803,21 @@ class DataService {
 
     def transformObject(Object object, Map includeFields) {
         Map properties = [:]
+        def value = null
         includeFields.each { fieldName, property ->
-            def value = property.tokenize('.').inject(object) { v, k -> v?."$k" }
-            properties[fieldName] = value ?: ""
+            if (property instanceof LinkedHashMap) {
+                value = property.value.tokenize('.').inject(object) { v, k -> v?."$k" }
+                if (property.format && value) {
+                    value = value.format(property.format)
+                }
+                if (property.defaultValue && !value) {
```

Comment:
We should support a property that has a default value and requires a format ... 
```[propertyName: "someDomain.someDateProperty", default: new Date(), format: "MM/dd/yyyy"]```

Reply:
That's right, I will add it